internal/tokens/fftokens/fftokens.go

```diff
@@ -477,6 +477,27 @@ func (ft *FFTokens) eventLoop() {
 	}
 }
 
+// Parse a JSON error of the form:
+//   {"error": "Bad Request", "message": "Field 'x' is required"}
+// into a message of the form:
+//   "Bad Request: Field 'x' is required"
+func wrapError(ctx context.Context, res *resty.Response, err error) error {
```

Comment:
There is a `SetError` function on the Resty client, where you can pass in a `fftypes.RESTError` object (or a more customized one) as a way to not need some of this complexity. Did you see that as an option?

Reply:
Did not know about that - I'll give it a look. I do want to fall back to the old behavior if the error doesn't conform to the expected JSON format from the connector though (ie in case nginx or something hops in the middle and spits back some text).